src/app/mod.rs

```diff
@@ -15,9 +36,20 @@ use tower_http::trace::TraceLayer;
 /// This panics upon failed to bind to port or if axum fails to serve app.
 ///
 pub async fn run() {
+    let app_state = match State::new().await {
+        Ok(value) => {
+            tracing::info!("Connected to database");
+            value
+        }
+        Err(e) => {
+            tracing::error!(error = %e, "Failed to connect to database");
+            panic!("Failed to connect to database");
+        }
+    };
```

Comment:
It would be nice to have this function panic itself so that we can reduce the pattern matching inside of the run function. Or just impl a .unwrap_or_else as shown in the config.rs file. Ill add a tracing on panic later so that we can just do .unwrap and it will properly trace it

Reply:
hmm i don't think it should panic itself, i think keeping it all at the same lever would be better

Reply:
It should still verbose and so remove the error matching and move it into an unwrap or with the success log being after

Reply:
```rust
let app_state = match State::new()
    .await
    .unwrap_or_else(|e| {
        tracing::error!(error = %e, "Failed to connect to database");
        panic!("Failed to connect to database");
    });
        
tracing::info!("Connected to database");
```

Reply:
This is what I ment